tests/rptest/tests/read_replica_e2e_test.py

```diff
@@ -72,6 +74,52 @@ def start_producer(self):
             message_validator=is_int_with_prefix)
         self.producer.start()
 
+    @cluster(num_nodes=6)
+    @matrix(partition_count=[10])
+    def test_produce_is_forbidden(self, partition_count):
+        # Create original topic
+        self.start_redpanda(3, si_settings=self.si_settings)
+        spec = TopicSpec(name=self.topic_name,
+                         partition_count=partition_count,
+                         replication_factor=3)
+
+        DefaultClient(self.redpanda).create_topic(spec)
+
+        # Make original topic upload data to S3
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(spec.name, 'redpanda.remote.write', 'true')
+
+        self.start_second_cluster()
+
+        def create_read_replica_topic_success():
+            try:
+                self.create_read_replica_topic()
+            except RpkException as e:
+                if "The server experienced an unexpected error when processing the request" in str(
+                        e):
+                    return False
+                else:
+                    raise
```

Comment:
nit: there is a flag you can pass to `wait_until` that will cause it to treat exceptions as False so you don't have to do this sort of thing.

Reply:
now it treats only the expected exception as False and raises for any other exception, so I'll leave it as is